packages/core/admin/admin/src/pages/Home/HomePage.tsx

```diff
@@ -54,6 +56,73 @@ export const WidgetRoot = ({ title, icon = PuzzlePiece, children }: RootProps) =
   );
 };
 
+/* -------------------------------------------------------------------------------------------------
+ * UnstableHomePageCe
+ * -----------------------------------------------------------------------------------------------*/
+
+const WidgetComponent = ({ component }: { component: () => Promise<React.ComponentType> }) => {
+  const [loadedComponent, setLoadedComponent] = React.useState<React.ComponentType | null>(null);
+
+  React.useEffect(() => {
+    const loadComponent = async () => {
+      const resolvedComponent = await component();
+
+      setLoadedComponent(() => resolvedComponent);
```

Comment:
I was a bit confused why I needed to provide a new function but without it the resolvedComponent function is invoked immediately throws warnings for violating the rules of hooks and then a bunch of errors

Reply:
storing components is always a mess, I guess the alternative may have been to use the ReactNode type and to display it with `{Component}` instead of `<Component />`

what you have is fine though!

Reply:
If you think this is acceptable I kinda prefer the <Component /> syntax even it is completely an aesthetic thing